newscoop/src/Newscoop/NewscoopBundle/EventListener/LocaleListener.php

```diff
@@ -47,62 +55,60 @@ public function onResponse(FilterResponseEvent $event)
     public function onRequest(GetResponseEvent $event)
     {
         $request = $event->getRequest();
-        $pos = strpos($request->server->get('REQUEST_URI'), '/admin');
+        $isAdmin = (strpos($request->server->get('REQUEST_URI'), '/admin') !== false);
         $cookies = $request->cookies;
+        $languageCode = null;
 
-        if ($cookies->has('TOL_Language')) {
-            $request->setLocale($cookies->get("TOL_Language"));
+        // Session language
+        if ($this->session->has('language')) {
```

Comment:
we need to compare this locale to other (from cookie, or from query - if they exists) and if they are not equal - then session one should be changed.